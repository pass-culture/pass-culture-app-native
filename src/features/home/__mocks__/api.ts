import { beneficiaryUser } from 'fixtures/user'

export const useUserProfileInfo = jest.fn().mockReturnValue({
  data: beneficiaryUser,
})
