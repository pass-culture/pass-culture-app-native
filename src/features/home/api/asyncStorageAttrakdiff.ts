import AsyncStorage from '@react-native-async-storage/async-storage'

import { Attrakdiff } from './useAttrakdiffModal'

export const createAsyncStorageAttrakdiff = (key: string): Attrakdiff => {
  return {
    hasTriggered: async () => (await AsyncStorage.getItem(key)) === 'true',
    setTriggered: async () => {
      await AsyncStorage.setItem(key, 'true')
    },
  }
}
