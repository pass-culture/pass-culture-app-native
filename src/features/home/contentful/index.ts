export * from './contentful'
export { ExclusivityPane, BusinessPane, Offers, OffersWithCover, VenuesModule } from './moduleTypes'
export { processHomepageEntry } from './processHomepageEntry'
