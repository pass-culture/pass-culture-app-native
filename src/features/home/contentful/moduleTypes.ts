import {
  ExclusivityDisplayParametersFields,
  DisplayParametersFields,
  SearchParametersFields,
  VenuesSearchParametersFields,
} from './contentful'

export class Offers {
  search: SearchParametersFields[]
  display: DisplayParametersFields
  moduleId: string
  constructor({
    search,
    display,
    moduleId,
  }: {
    search: SearchParametersFields[]
    display: DisplayParametersFields
    moduleId: string
  }) {
    this.search = search
    this.display = display
    this.moduleId = moduleId
  }
}

export class OffersWithCover extends Offers {
  cover: string | null
  constructor({
    cover,
    display,
    search,
    moduleId,
  }: {
    search: SearchParametersFields[]
    display: DisplayParametersFields
    cover: string | null
    moduleId: string
  }) {
    super({ search, display, moduleId })
    this.cover = cover
  }
}

export class VenuesModule {
  search: VenuesSearchParametersFields[]
  display: DisplayParametersFields
  moduleId: string
  constructor({
    search,
    display,
    moduleId,
  }: {
    search: VenuesSearchParametersFields[]
    display: DisplayParametersFields
    moduleId: string
  }) {
    this.search = search
    this.display = display
    this.moduleId = moduleId
  }
}

export class ExclusivityPane {
  alt: string
  image: string
  id: number
  moduleId: string
  display?: ExclusivityDisplayParametersFields
  constructor({
    alt,
    image,
    id,
    moduleId,
    display,
  }: {
    alt: string
    image: string
    id: number
    moduleId: string
    display?: ExclusivityDisplayParametersFields
  }) {
    this.alt = alt
    this.image = image
    this.id = id
    this.moduleId = moduleId
    this.display = display
  }
}

export class RecommendationPane {
  display: DisplayParametersFields
  constructor({ display }: { display: DisplayParametersFields }) {
    this.display = display
  }
}

export class BusinessPane {
  title: string
  firstLine: string | undefined
  image: string
  leftIcon: string | undefined
  secondLine: string | undefined
  url: string | undefined
  moduleId: string
  targetNotConnectedUsersOnly: boolean | undefined
  constructor({
    title,
    firstLine,
    image,
    leftIcon,
    secondLine,
    url,
    moduleId,
    targetNotConnectedUsersOnly,
  }: {
    title: string
    firstLine: string | undefined
    image: string
    leftIcon: string | undefined
    secondLine: string | undefined
    url: string | undefined
    moduleId: string
    targetNotConnectedUsersOnly: boolean | undefined
  }) {
    this.title = title
    this.firstLine = firstLine
    this.image = image
    this.leftIcon = leftIcon
    this.secondLine = secondLine
    this.url = url
    this.moduleId = moduleId
    this.targetNotConnectedUsersOnly = targetNotConnectedUsersOnly
  }
}

export type ProcessedModule =
  | Offers
  | ExclusivityPane
  | BusinessPane
  | OffersWithCover
  | RecommendationPane
  | VenuesModule
