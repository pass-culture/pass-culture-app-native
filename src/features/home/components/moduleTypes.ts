import { AlgoliaParametersFields, DisplayParametersFields } from 'features/home/contentful.d'

export class Offers {
  algolia: AlgoliaParametersFields
  display: DisplayParametersFields
  constructor({
    algolia,
    display,
  }: {
    algolia: AlgoliaParametersFields
    display: DisplayParametersFields
  }) {
    this.algolia = algolia
    this.display = display
  }
}

export class OffersWithCover extends Offers {
  cover: string | null
  constructor({
    algolia,
    cover,
    display,
  }: {
    algolia: AlgoliaParametersFields
    display: DisplayParametersFields
    cover: string | null
  }) {
    super({ algolia, display })
    this.cover = cover
  }
}

export class ExclusivityPane {
  alt: string
  image: string
  offerId: string
  constructor({ alt, image, offerId }: { alt: string; image: string; offerId: string }) {
    this.alt = alt
    this.image = image
    this.offerId = offerId
  }
}

export class BusinessPane {
  firstLine: string | undefined
  image: string | undefined | null
  secondLine: string | undefined
  url: string | undefined
  constructor({
    firstLine,
    image,
    secondLine,
    url,
  }: {
    firstLine: string | undefined
    image: string | undefined | null
    secondLine: string | undefined
    url: string | undefined
  }) {
    this.firstLine = firstLine
    this.image = image
    this.secondLine = secondLine
    this.url = url
  }
}

export type ProcessedModule = Offers | ExclusivityPane | BusinessPane | OffersWithCover
