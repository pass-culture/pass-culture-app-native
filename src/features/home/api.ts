import resolveResponse from 'contentful-resolve-response'
import { useMemo, useEffect } from 'react'
import { useQuery } from 'react-query'

import { api } from 'api/api'
import { UserProfileResponse } from 'api/gen'
import { useAuthContext } from 'features/auth/AuthContext'
import { NoContentError } from 'features/home/components/NoContentError'
import {
  HomepageEntry,
  EntryCollection,
  EntryFields,
  processHomepageEntry,
} from 'features/home/contentful'
import { useSelectPlaylist } from 'features/home/selectPlaylist'
import { analytics } from 'libs/analytics'
import { env } from 'libs/environment'
import { getExternal } from 'libs/fetch'
import { ScreenError } from 'libs/monitoring'
import { useNetwork } from 'libs/network/useNetwork'
import { QueryKeys } from 'libs/queryKeys'

const DEPTH_LEVEL = 2

export const CONTENTFUL_BASE_URL = 'https://cdn.contentful.com'
export const BASE_URL = `${CONTENTFUL_BASE_URL}/spaces/${env.CONTENTFUL_SPACE_ID}/environments/${env.CONTENTFUL_ENVIRONMENT}`
export const PARAMS = `?include=${DEPTH_LEVEL}&content_type=homepageNatif&access_token=${env.CONTENTFUL_ACCESS_TOKEN}`

export async function getEntries() {
  const url = `${BASE_URL}/entries${PARAMS}`
  try {
    const json = await getExternal<EntryCollection<EntryFields, 'homepageNatif'>>(url)
    return resolveResponse(json)
  } catch (e) {
    const error = e as Error
    throw new ScreenError(error?.message, NoContentError)
  }
}

export function useHomepageModules(paramsEntryId?: string) {
  const { isConnected } = useNetwork()
  const selectPlaylist = useSelectPlaylist(paramsEntryId)
  const { data: entries } = useQuery<HomepageEntry[]>(QueryKeys.HOMEPAGE_MODULES, getEntries, {
    enabled: isConnected,
  })

  const entry = selectPlaylist(entries || [])
  const entryId = entry?.sys.id

  useEffect(() => {
    if (entryId) analytics.logConsultHome({ entryId })
  }, [entryId])

  return useMemo(() => (entry ? processHomepageEntry(entry) : []), [entryId])
}

export function useUserProfileInfo(options = {}) {
  const { isConnected } = useNetwork()
  const { isLoggedIn } = useAuthContext()

  return useQuery<UserProfileResponse>(QueryKeys.USER_PROFILE, () => api.getnativev1me(), {
    enabled: isLoggedIn && isConnected,
    ...options,
  })
}
