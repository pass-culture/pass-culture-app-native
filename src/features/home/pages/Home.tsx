import { t } from '@lingui/macro'
import { useFocusEffect, useNavigation, useRoute } from '@react-navigation/native'
import React, { FunctionComponent, useState } from 'react'
import { NativeSyntheticEvent, NativeScrollEvent, ScrollView, TouchableOpacity } from 'react-native'
import { getStatusBarHeight } from 'react-native-iphone-x-helper'
import styled from 'styled-components/native'

import { useUserProfileInfo } from 'features/home/api'
import { HomeBodyPlaceholder, SignUpSignInChoiceModal } from 'features/home/components'
import { HomeBody } from 'features/home/components/HomeBody'
import { useDisplayedHomeModules } from 'features/home/pages/useDisplayedHomeModules'
import { useAvailableCredit } from 'features/home/services/useAvailableCredit'
import { UseNavigationType, UseRouteType } from 'features/navigation/RootNavigator'
import { analytics } from 'libs/analytics'
import { isCloseToBottom } from 'libs/analytics.utils'
import { env } from 'libs/environment'
import { _ } from 'libs/i18n'
import { formatToFrenchDecimal } from 'libs/parsers'
import { useModal } from 'ui/components/modals/useModal'
import { HeaderBackground } from 'ui/svg/HeaderBackground'
import { UserCircle } from 'ui/svg/icons/UserCircle'
import { ColorsEnum, getSpacing, Spacer, Typo } from 'ui/theme'
import { ACTIVE_OPACITY } from 'ui/theme/colors'
import { ZIndexes } from 'ui/theme/layers'

import { useShowSkeleton } from './useShowSkeleton'

const statusBarHeight = getStatusBarHeight(true)

export const Home: FunctionComponent = function () {
  const navigation = useNavigation<UseNavigationType>()
  const { params } = useRoute<UseRouteType<'Home'>>()
  const { data: userInfos } = useUserProfileInfo()
  const { visible: signInModalVisible, showModal: showSignInModal, hideModal } = useModal(false)
  const showSkeleton = useShowSkeleton()
  const availableCredit = useAvailableCredit()

  const [hasSeenAllModules, setHasSeenAllModules] = useState<boolean>(false)
  const { displayedModules, algoliaModules } = useDisplayedHomeModules()

  function hideSignInModal() {
    navigation.setParams({ shouldDisplayLoginModal: false })
    hideModal()
  }

  useFocusEffect(() => {
    if (params.shouldDisplayLoginModal) {
      showSignInModal()
    }
  })

  const checkIfAllModulesHaveBeenSeen = ({
    nativeEvent,
  }: NativeSyntheticEvent<NativeScrollEvent>) => {
    if (!hasSeenAllModules && isCloseToBottom(nativeEvent)) {
      setHasSeenAllModules(true)
      analytics.logAllModulesSeen(displayedModules.length)
    }
  }

  let subtitle = _(t`Toute la culture dans ta main`)
  if (availableCredit) {
    subtitle = availableCredit.isExpired
      ? _(t`Ton crédit est expiré`)
      : _(t`Tu as ${formatToFrenchDecimal(availableCredit.amount)} sur ton pass`)
  }

  return (
    <ScrollView
      testID="homeScrollView"
      scrollEventThrottle={400}
      bounces={false}
      onScroll={checkIfAllModulesHaveBeenSeen}>
      <Spacer.TopScreen />
      {env.CHEAT_BUTTONS_ENABLED && (
        <CheatButtonsContainer>
          <CheatTouchableOpacity onPress={() => navigation.navigate('AppComponents')}>
            <Typo.Body>{_(t`Composants`)}</Typo.Body>
          </CheatTouchableOpacity>
          <CheatTouchableOpacity onPress={() => navigation.navigate('Navigation')}>
            <Typo.Body>{_(t`Navigation`)}</Typo.Body>
          </CheatTouchableOpacity>
        </CheatButtonsContainer>
      )}
      <HeaderBackgroundWrapper>
        <HeaderBackground />
      </HeaderBackgroundWrapper>
      <UserProfileContainer onPress={() => navigation.navigate('CheatMenu')}>
        <UserCircle size={32} color={ColorsEnum.WHITE} />
      </UserProfileContainer>

      <CenterContainer>
        <Spacer.Column numberOfSpaces={8} />
        <StyledTitle1 color={ColorsEnum.WHITE} numberOfLines={2}>
          {userInfos?.firstName
            ? _(/*i18n: Hello title message */ t`Bonjour ${userInfos?.firstName}`)
            : _(/*i18n: Welcome title message */ t`Bienvenue !`)}
        </StyledTitle1>
        <Spacer.Column numberOfSpaces={2} />
        <Typo.Body color={ColorsEnum.WHITE}>{subtitle}</Typo.Body>
      </CenterContainer>
      <Spacer.Column numberOfSpaces={6} />

      {showSkeleton ? <HomeBodyPlaceholder /> : null}
      <HomeBodyLoadingContainer isLoading={showSkeleton}>
        <HomeBody modules={displayedModules} algoliaModules={algoliaModules} />
      </HomeBodyLoadingContainer>

      <SignUpSignInChoiceModal visible={signInModalVisible} dismissModal={hideSignInModal} />
      <Spacer.TabBar />
    </ScrollView>
  )
}

const StyledTitle1 = styled(Typo.Title1)({
  textAlign: 'center',
  marginHorizontal: getSpacing(8),
})

const CenterContainer = styled.View({
  flex: 1,
  alignItems: 'center',
})

const HeaderBackgroundWrapper = styled.View({
  position: 'absolute',
  top: 0,
  left: 0,
  zIndex: 0,
})

const HomeBodyLoadingContainer = styled.View<{ isLoading: boolean }>(({ isLoading }) => ({
  height: isLoading ? 0 : undefined,
  overflow: 'hidden',
}))

const CheatButtonsContainer = styled.View({
  width: '100%',
  flexDirection: 'row',
  justifyContent: 'space-around',
  zIndex: ZIndexes.HomeCheatButton,
  top: 20,
})

const UserProfileContainer = styled.TouchableOpacity.attrs({
  activeOpacity: ACTIVE_OPACITY,
})({
  position: 'absolute',
  right: 24,
  top: getSpacing(3) + statusBarHeight,
  zIndex: 1,
})

const CheatTouchableOpacity = styled(TouchableOpacity)({
  borderColor: ColorsEnum.BLACK,
  borderWidth: 2,
  padding: getSpacing(1),
})
