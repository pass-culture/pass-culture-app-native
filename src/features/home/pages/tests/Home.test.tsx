import { render, act } from '@testing-library/react-native'
import { rest } from 'msw'
import React from 'react'
import waitForExpect from 'wait-for-expect'

import { useRoute } from '__mocks__/@react-navigation/native'
import { UserProfileResponse } from 'api/gen'
import { AuthContext } from 'features/auth/AuthContext'
import { ProcessedModule } from 'features/home/contentful'
import { RecommendationPane } from 'features/home/contentful/moduleTypes'
import { mockedAlgoliaResponse } from 'libs/algolia/mockedResponses/mockedAlgoliaResponse'
import { analytics } from 'libs/analytics'
import { env } from 'libs/environment'
import { reactQueryProviderHOC } from 'tests/reactQueryProviderHOC'
import { server } from 'tests/server'
import { flushAllPromises, superFlushWithAct } from 'tests/utils'

import { Home } from '../Home'

jest.mock('libs/environment', () => ({
  env: {
    FEATURE_FLIPPING_ONLY_VISIBLE_ON_TESTING: false,
  },
}))

jest.mock('features/home/pages/useShowSkeleton', () => ({
  useShowSkeleton: jest.fn(() => false),
}))

let mockDisplayedModules: ProcessedModule[] = []
jest.mock('features/home/pages/useDisplayedHomeModules', () => ({
  useDisplayedHomeModules: jest.fn(() => ({
    displayedModules: mockDisplayedModules,
    algoliaModules: {},
    recommendedHits: mockedAlgoliaResponse.hits.slice(0, 4),
  })),
}))

function simulateAuthenticatedUser(partialUser?: Partial<UserProfileResponse>) {
  server.use(
    rest.get(env.API_BASE_URL + '/native/v1/me', (req, res, ctx) => {
      return res(
        ctx.status(200),
        ctx.json({
          email: 'email@domain.ext',
          firstName: 'Jean',
          depositExpirationDate: '2023-02-16T17:16:04.735235',
          domainsCredit: {
            all: { initial: 50000, remaining: 49600 },
            physical: { initial: 20000, remaining: 0 },
            digital: { initial: 20000, remaining: 19600 },
          },
          ...(partialUser || {}),
        } as UserProfileResponse)
      )
    })
  )
}

describe('Home component', () => {
  it('should render correctly without login modal', async () => {
    env.FEATURE_FLIPPING_ONLY_VISIBLE_ON_TESTING = false
    const home = await homeRenderer({ isLoggedIn: false, withModal: false })
    expect(home).toMatchSnapshot()
  })

  it('should render modal correctly', async () => {
    const { getByText } = await homeRenderer({ isLoggedIn: false, withModal: true })
    expect(getByText('Se connecter')).toBeTruthy()
  })

  it('should have a welcome message', async () => {
    const { getByText } = await homeRenderer({ isLoggedIn: false, withModal: false })
    const welcomeText = getByText('Bienvenue !')
    expect(welcomeText.props.children).toBe('Bienvenue !')
  })

  it('should have a personalized welcome message when user is logged in', async () => {
    const { getByText } = await homeRenderer({ isLoggedIn: true, withModal: false })
    await waitForExpect(() => {
      expect(getByText('Bonjour Jean')).toBeTruthy()
    })
  })

  it('should show the available credit to the user - remaining', async () => {
    const { getByText } = await homeRenderer({ isLoggedIn: true, withModal: false })
    await waitForExpect(() => {
      expect(getByText('Tu as 496 € sur ton pass')).toBeTruthy()
    })
  })

  it('should show the available credit to the user - expired', async () => {
    const { queryByText, getByText } = await homeRenderer({
      isLoggedIn: true,
      withModal: false,
      partialUser: { depositExpirationDate: new Date('2020-02-16T17:16:04.735235') },
    })
    await waitForExpect(() => {
      expect(getByText('Ton crédit est expiré')).toBeTruthy()
      expect(queryByText('Tu as 496 € sur ton pass')).toBeFalsy()
    })
  })

  it('should show the available credit to the user - not logged in', async () => {
    const { queryByText } = await homeRenderer({ isLoggedIn: false, withModal: false })
    expect(queryByText('Toute la culture dans ta main')).toBeTruthy()
  })

  it('should not have code push button', async () => {
    const home = await homeRenderer({ withModal: false })
    expect(home.queryByText('Check update')).toBeFalsy()
  })

  it('should have CheatMenu button when FEATURE_FLIPPING_ONLY_VISIBLE_ON_TESTING=true', async () => {
    env.FEATURE_FLIPPING_ONLY_VISIBLE_ON_TESTING = true
    const home = await homeRenderer({ isLoggedIn: false, withModal: false })
    expect(home.queryByText('CheatMenu')).toBeTruthy()
  })

  it('should NOT have CheatMenu button when NOT FEATURE_FLIPPING_ONLY_VISIBLE_ON_TESTING=false', async () => {
    env.FEATURE_FLIPPING_ONLY_VISIBLE_ON_TESTING = false
    const home = await homeRenderer({ isLoggedIn: false, withModal: false })
    expect(home.queryByText('CheatMenu')).toBeFalsy()
  })
})

describe('Home component - Analytics', () => {
  const nativeEventMiddle = {
    layoutMeasurement: { height: 1000 },
    contentOffset: { y: 400 }, // how far did we scroll
    contentSize: { height: 1600 },
  }
  const nativeEventBottom = {
    layoutMeasurement: { height: 1000 },
    contentOffset: { y: 900 },
    contentSize: { height: 1600 },
  }

  it('should trigger logEvent "AllModulesSeen" when reaching the end', async () => {
    const home = await homeRenderer({ isLoggedIn: false, withModal: false })
    const scrollView = home.getByTestId('homeScrollView')
    await act(async () => {
      await flushAllPromises()
    })

    await act(async () => {
      await scrollView.props.onScroll({ nativeEvent: nativeEventMiddle })
    })
    expect(analytics.logAllModulesSeen).not.toHaveBeenCalled()

    await act(async () => {
      await scrollView.props.onScroll({ nativeEvent: nativeEventBottom })
    })

    expect(analytics.logAllModulesSeen).toHaveBeenCalledWith(0)
  })

  it('should trigger logEvent "AllModulesSeen" only once', async () => {
    const home = await homeRenderer({ isLoggedIn: false, withModal: false })
    const scrollView = home.getByTestId('homeScrollView')

    await act(async () => {
      await flushAllPromises()
    })

    await act(async () => {
      // 1st scroll to bottom => trigger
      await scrollView.props.onScroll({ nativeEvent: nativeEventBottom })
      await flushAllPromises()
    })
    expect(analytics.logAllModulesSeen).toHaveBeenCalledWith(0)

    // @ts-ignore: logAllModulesSeen is the mock function but is seen as the real function
    analytics.logAllModulesSeen.mockClear()

    await act(async () => {
      // 2nd scroll to bottom => NOT trigger
      await scrollView.props.onScroll({ nativeEvent: nativeEventMiddle })
      await scrollView.props.onScroll({ nativeEvent: nativeEventBottom })
      await flushAllPromises()
    })

    expect(analytics.logAllModulesSeen).not.toHaveBeenCalled()
  })

  it('should trigger logEvent "RecommendationModuleSeen" when reaching the recommendation module', async () => {
    mockDisplayedModules = [
      new RecommendationPane({
        display: { title: 'Tes offres recommandées', minOffers: 2, layout: 'one-item-medium' },
      }),
    ]
    const home = await homeRenderer({ isLoggedIn: false, withModal: false })
    const scrollView = home.getByTestId('homeScrollView')

    await act(async () =>
      home
        .getByTestId('recommendationModuleTracker')
        .props.onLayout({ nativeEvent: { layout: { y: 1500 } } })
    )
    expect(home.getByTestId('recommendationModuleTracker')).toBeTruthy()

    await act(async () => {
      await scrollView.props.onScroll({ nativeEvent: nativeEventMiddle })
    })
    expect(analytics.logRecommendationModuleSeen).not.toHaveBeenCalled()

    await act(async () => {
      await scrollView.props.onScroll({ nativeEvent: nativeEventBottom })
    })
    expect(analytics.logRecommendationModuleSeen).toHaveBeenCalledWith('Tes offres recommandées', 4)
  })
})

interface Props {
  isLoggedIn?: boolean | undefined
  withModal?: boolean | undefined
  partialUser?: Partial<UserProfileResponse>
}

async function homeRenderer(
  { isLoggedIn, withModal, partialUser }: Props = {
    isLoggedIn: false,
    withModal: false,
    partialUser: {},
  }
) {
  useRoute.mockReturnValue({
    params: {
      shouldDisplayLoginModal: withModal,
    },
  })
  if (isLoggedIn) simulateAuthenticatedUser(partialUser)
  const renderAPI = render(
    reactQueryProviderHOC(
      <AuthContext.Provider value={{ isLoggedIn: !!isLoggedIn, setIsLoggedIn: jest.fn() }}>
        <Home />
      </AuthContext.Provider>
    )
  )
  await superFlushWithAct(50)
  return renderAPI
}
