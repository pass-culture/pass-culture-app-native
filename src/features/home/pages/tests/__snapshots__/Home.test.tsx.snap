// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render correctly without login modal 1`] = `
<RCTScrollView
  bounces={false}
  onScroll={[Function]}
  scrollEventThrottle={400}
  testID="homeScrollView"
>
  <View>
    <View
      customHeight={16}
      style={
        Array [
          Object {
            "height": 16,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "left": 0,
            "position": "absolute",
            "top": 0,
            "zIndex": 0,
          },
        ]
      }
    >
      <View
        height={280}
        width={754}
      >
        <Text>
          undefined-SVG-Mock
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <View
        numberOfSpaces={8}
        style={
          Array [
            Object {
              "height": 32,
            },
          ]
        }
      />
      <Text
        color="#ffffff"
        numberOfLines={2}
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Montserrat-ExtraBoldItalic",
              "fontSize": 28,
              "lineHeight": 34,
              "marginHorizontal": 32,
              "textAlign": "center",
            },
          ]
        }
      >
        Bienvenue !
      </Text>
      <View
        numberOfSpaces={2}
        style={
          Array [
            Object {
              "height": 8,
            },
          ]
        }
      />
      <Text
        color="#ffffff"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 15,
              "lineHeight": 20,
            },
          ]
        }
      >
        Toute la culture dans ta main
      </Text>
    </View>
    <View
      numberOfSpaces={6}
      style={
        Array [
          Object {
            "height": 24,
          },
        ]
      }
    />
    <View
      isLoading={false}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        <View
          numberOfSpaces={6}
          style={
            Array [
              Object {
                "height": 24,
              },
            ]
          }
        />
      </View>
    </View>
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      statusBarTranslucent={true}
      testID="modal"
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "column",
            "flexGrow": 1,
            "height": "100%",
            "justifyContent": "flex-end",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="click-away-area"
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderTopEndRadius": 16,
              "borderTopStartRadius": 16,
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "maxHeight": "85%",
              "opacity": 1,
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "100%",
                  "zIndex": 1,
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "flex-start",
                  "flexBasis": 0,
                  "flexGrow": 0.1,
                  "flexShrink": 1,
                  "marginLeft": -4,
                  "opacity": 1,
                }
              }
              testID="leftIconButton"
            />
            <Text
              numberOfLines={2}
              style={
                Array [
                  Object {
                    "color": "#151515",
                    "flexBasis": 0,
                    "flexGrow": 0.8,
                    "flexShrink": 1,
                    "fontFamily": "Montserrat-Medium",
                    "fontSize": 18,
                    "lineHeight": 22,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Le pass Culture est accessible à tous !
            </Text>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "flex-end",
                  "flexBasis": 0,
                  "flexGrow": 0.1,
                  "flexShrink": 1,
                  "marginRight": -4,
                  "opacity": 1,
                }
              }
              testID="rightIconButton"
            >
              <View
                height={32}
                testID="rightIcon"
                width={32}
              >
                <Text>
                  rightIcon-SVG-Mock
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "maxWidth": 500,
                  "paddingTop": 20,
                  "width": "100%",
                },
                Object {
                  "paddingBottom": 8,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "paddingBottom": 30,
                    "textAlign": "center",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#151515",
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 15,
                      "lineHeight": 20,
                    },
                  ]
                }
              >
                Si tu as 18 ans, tu es éligible pour obtenir une aide financière de 300 € proposée par le Ministère de la Culture qui sera créditée directement sur ton compte pass Culture.
              </Text>
            </Text>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#eb0055",
                  "borderRadius": 24,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "center",
                  "maxWidth": 500,
                  "opacity": 1,
                  "paddingBottom": 2,
                  "paddingLeft": 2,
                  "paddingRight": 2,
                  "paddingTop": 2,
                  "width": "100%",
                }
              }
              testID="button-container"
            >
              <Text
                adjustsFontSizeToFit={false}
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Montserrat-Bold",
                      "fontSize": 15,
                      "lineHeight": 20,
                      "marginLeft": 5,
                      "maxWidth": 650,
                    },
                  ]
                }
                testID="button-title"
                textColor="#ffffff"
              >
                S'inscrire
              </Text>
            </View>
            <View
              numberOfSpaces={3}
              style={
                Array [
                  Object {
                    "height": 12,
                  },
                ]
              }
            />
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 24,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "center",
                  "maxWidth": 500,
                  "opacity": 1,
                  "paddingBottom": 2,
                  "paddingLeft": 2,
                  "paddingRight": 2,
                  "paddingTop": 2,
                  "width": "100%",
                }
              }
              testID="button-container"
            >
              <Text
                adjustsFontSizeToFit={false}
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#eb0055",
                      "fontFamily": "Montserrat-Bold",
                      "fontSize": 15,
                      "lineHeight": 20,
                      "marginLeft": 5,
                      "maxWidth": 650,
                    },
                  ]
                }
                testID="button-title"
                textColor="#eb0055"
              >
                Se connecter
              </Text>
            </View>
          </View>
        </View>
      </View>
    </Modal>
    <View
      customHeight={72}
      style={
        Array [
          Object {
            "height": 72,
          },
        ]
      }
    />
  </View>
</RCTScrollView>
`;
